### PR TITLE
[API 2.0] Fix api 'is_empty' 

### PR DESCRIPTION
**Pull request by danleifeng**
### PR types

Bug fixes
### PR changes

 APIs
### Describe

fix is_empty, main changes include:
1. api interface from  `is_empty(x, cond=None)` to `is_empty(x, name=None)`
2. add dygraph mode for `is_empty`
3. edit example code in static mode
4. add example code in dygraph mode

image

chinese doc PR: https://github.com/PaddlePaddle/FluidDoc/pull/2692

### CODE REVIEW

File: python/paddle/fluid/layers/control_flow.py [hunk 1]
```diff
@@ -3788,45 +3788,67 @@ def reorder_lod_tensor_by_rank(x, rank_table):
     return out
 
 
-def is_empty(x, cond=None):
+def is_empty(x, name=None):
     """
     :alias_main: paddle.is_empty
 	:alias: paddle.is_empty,paddle.tensor.is_empty,paddle.tensor.logic.is_empty
 	:old_api: paddle.fluid.layers.is_empty
```
Comment (TCChenlong) [inline python/paddle/fluid/layers/control_flow.py]: 上面三行的内容可以删除

File: python/paddle/fluid/layers/control_flow.py [hunk 2]
```diff
@@ -3788,45 +3788,67 @@ def reorder_lod_tensor_by_rank(x, rank_table):
...
+            print("res:", res)
+            # ('res:', Tensor: eager_tmp_1
+            #    - place: CPUPlace
+            #    - shape: [1]
+            #    - layout: NCHW
+            #    - dtype: bool
+            #    - data: [0])
+
+
+        .. code-block:: python
```
Comment (TCChenlong) [inline python/paddle/fluid/layers/control_flow.py]: 动态图的示例即可 不需要提供静态图的示例

File: python/paddle/fluid/layers/control_flow.py (hunk 1 above)
Comment (danleifeng) [reply python/paddle/fluid/layers/control_flow.py]: done,thanks

File: python/paddle/fluid/layers/control_flow.py (hunk 2 above)
Comment (danleifeng) [reply python/paddle/fluid/layers/control_flow.py]: done,thanks

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by TCChenlong — APPROVED** [review]
LGTM

**Review by kolinwei — APPROVED** [review]

**Pull request merged**